### PR TITLE
added tray popupcontextmenu(#373)

### LINKED ISSUE #373: Tray context menu popup
Hello, 
I'm developing a windows application with go-astilectron and i couldn't find a proper way to open tray context menu on click event or manually, so is there any way to open tray context menu?
This action could be done in electron by below link
https://www.electronjs.org/docs/latest/api/tray#traypopupcontextmenumenu-position-macos-windows

Thanks

### PR DESCRIPTION
**Pull request by worlpaker**
Adds:
- Tray PopupContextMenu

Fixes #373

Thanks

### CODE REVIEW

File: tray.go [hunk 1]
```diff
@@ -8,15 +8,17 @@ import (
...
-	EventNameTrayEventClicked       = "tray.event.clicked"
-	EventNameTrayEventCreated       = "tray.event.created"
-	EventNameTrayEventDestroyed     = "tray.event.destroyed"
-	EventNameTrayEventDoubleClicked = "tray.event.double.clicked"
-	EventNameTrayEventImageSet      = "tray.event.image.set"
-	EventNameTrayEventRightClicked  = "tray.event.right.clicked"
+	EventNameTrayCmdCreate                = "tray.cmd.create"
+	EventNameTrayCmdDestroy               = "tray.cmd.destroy"
+	EventNameTrayCmdSetImage              = "tray.cmd.set.image"
+	EventNameTrayCmdPopUpContextMenu      = "tray.cmd.popup.contextmenu"
```
Comment (asticode) [inline tray.go:14]: Could you replace this with:

`EventNameTrayCmdPopupContextMenu      = "tray.cmd.popup.context.menu"`

File: tray.go [hunk 2]
```diff
@@ -8,15 +8,17 @@ import (
...
+	EventNameTrayEventClicked             = "tray.event.clicked"
+	EventNameTrayEventCreated             = "tray.event.created"
+	EventNameTrayEventDestroyed           = "tray.event.destroyed"
+	EventNameTrayEventDoubleClicked       = "tray.event.double.clicked"
+	EventNameTrayEventImageSet            = "tray.event.image.set"
+	EventNameTrayEventRightClicked        = "tray.event.right.clicked"
+	EventNameTrayEventPoppedUpContextMenu = "tray.event.poppedup.contextmenu"
```
Comment (asticode) [inline tray.go:21]: Could you replace this with:

`EventNameTrayEventContextMenuPoppedUp = "tray.event.context.menu.popped.up"`

File: tray.go [hunk 3]
```diff
@@ -34,6 +36,12 @@ type TrayOptions struct {
 	Tooltip *string `json:"tooltip,omitempty"`
 }
 
+// TrayPopUpOptions represents Tray PopUpContextMenu options
+type TrayPopUpOptions struct {
+	Menu     *Menu
+	Position PositionOptions
```
Comment (asticode) [inline tray.go:42]: I think this should be a pointer as well so that it's optional

**Review by asticode — CHANGES_REQUESTED** [review]

File: tray.go (hunk 1 above)
Comment (worlpaker) [reply tray.go]: Done.

File: tray.go (hunk 2 above)
Comment (worlpaker) [reply tray.go]: Done

**Comment by asticode** [comment]
Thanks for the PR! ❤️ 

FYI I've created a v0.30.0 tag

**Comment by worlpaker** [comment]
Thanks.

But I think you missed 0 when creating a tag.

You created [v.30.0](https://github.com/asticode/go-astilectron/releases/tag/v.30.0)

which I think should be:

v0.30.0

**Comment by asticode** [comment]
You're right!

I've corrected the tag 👍 

Thanks!

**Pull request merged**